packages/hub/node-tests/@core/compiler-test.ts

```diff
@@ -484,6 +484,31 @@ if (process.env.COMPILER) {
         `);
       });
 
+      it('can compile schema class promise functions', async function () {
+        let { compiled } = await cards.load(`${realm}person`);
+        let source = getFileCache().getModule(compiled.schemaModule.global, 'browser');
+        expect(source).to.containsSource(`
+          then(resolve) {
```

Comment:
we can set the params for this to `resolve, reject` and thread the error that is raised from the Promise.all.catch thru to the reject